overlord/state/state.go

```diff
@@ -379,14 +385,16 @@ func (s *State) Prune(pruneWait, abortWait time.Duration) {
 			}
 			continue
 		}
-		if readyTime.Before(pruneLimit) {
+		// change old or we have too many changes
+		if readyTime.Before(pruneLimit) || (chg.Status().Ready() && len(s.changes) > maxReadyChanges) {
```

Comment:
Change statuses aren't super cheap to compute. Best to avoid it in a frequent global iteration like this.

There's also no need in this case I believe. It _has_ to be ready, otherwise readyTime would be zero and we wouldn't be here.

The logic here doesn't seem quite healthy though. This may end up killing changes as soon as they're done, to the point of the client being unable to even see their result.

Let's please talk about this.